### PR TITLE
Consider RETURN_TIME_AS_PERIOD for relative times

### LINKED ISSUE #806: RETURN_TIME_AS_PERIOD doesn't always work
For example I'd expect the following to have a period of time, but it's day:

```
>>> import dateparser
>>> dp = dateparser.date.DateDataParser(languages=['en'], settings={'RETURN_TIME_AS_PERIOD': True})
>>> dp.get_date_data('yesterday at 11:34')
{'date_obj': datetime.datetime(2020, 10, 7, 11, 34), 'period': 'day', 'locale': 'en'}
```

Whereas the following has a period of time as expected:

```
>>> dp.get_date_data('11:34')
{'date_obj': datetime.datetime(2020, 10, 8, 11, 34), 'period': 'time', 'locale': 'en'}
```

### PR DESCRIPTION
**Pull request by onlynone**
This should set the period to time when RETURN_TIME_AS_PERIOD is set, and you parse a date like 'yesterday at 1:30'. This should fix #806 .

### CODE REVIEW

File: dateparser/freshness_date_parser.py [hunk 1]
```diff
@@ -91,7 +94,7 @@ def apply_time(dateobj, timeobj):
         date, period = self._parse_date(date_string, settings.PREFER_DATES_FROM)
 
         if date:
-            date = apply_time(date, _time)
+            date, period = apply_time(date, _time, period)
```
Comment (Gallaecio) [inline dateparser/freshness_date_parser.py:97]: :lipstick: I personally think that checking if the return value is the same as the passed `date` object after this line, and changing `period` based on that, would look better than the proposed changes to the implementation of `apply_time`.

**Review by Gallaecio — APPROVED** [review]

File: dateparser/freshness_date_parser.py (hunk 1 above)
Comment (onlynone) [reply dateparser/freshness_date_parser.py]: Made this change.

**Comment by onlynone** [comment]
@Gallaecio I made the changes you requested, but I don't know why that made the test coverage go down.

**Review by noviluni — APPROVED** [review]
Nice!

**Pull request merged**